play27-json-ops/src/main/scala/play/api/libs/json/ops/JsResults.scala

```diff
@@ -0,0 +1,70 @@
+package play.api.libs.json.ops
+
+import play.api.libs.json.{JsError, JsPath, JsResult}
+
+import scala.collection.generic.CanBuildFrom
+import scala.language.higherKinds
+
+/**
+ * Useful helper methods for combining and filtering [[JsResult]]s.
+ */
+object JsResults {
+
+  /**
+   * Filter out all [[JsError]]s and return a collection of results with the most specific type.
+   *
+   * @param results the results to filter
+   * @return a collection of results in whatever order the given collection type maintains
+   */
+  def ignoreErrors[T, C[_]](results: Traversable[JsResult[T]])
+    (implicit bf: CanBuildFrom[Traversable[JsResult[T]], T, C[T]]): C[T] = {
+    results.filter(_.isSuccess).map(_.get)
+  }
+
+  /**
+   * Collects all [[JsError]]s and appends the index of result to path of the error.
+   *
+   * @param results a sequence of results
+   * @return a collection of the JsError objects in the sequence that they were given
+   */
+  def collectErrors[C[_]](results: Seq[JsResult[_]])
+    (implicit bf: CanBuildFrom[Seq[JsResult[_]], JsError, C[JsError]]): C[JsError] = {
+    results.zipWithIndex map {
+      case (result, index) => result.repath(JsPath(index))
+    } collect {
+      case error: JsError => error
```

Comment:
I should probably collect before repath to avoid wasting time repathing successful results.